### PR TITLE
Format sequences in bzl mode flexibly

### PR DESCRIPTION
**Pull request by vladmos**
Sequences are printed in a compact mode if there have been originally formatted in a compact mode.

### CODE REVIEW

File: build/print.go [hunk 1]
```diff
@@ -631,60 +631,81 @@ const (
...
+			if start.Line != previousEnd {
+				return false
+			}
+			previousEnd = end.Line
+		}
+		if end != nil && previousEnd != end.Pos.Line {
+			return false
+		}
+		return true
+	} else {
```
Comment (laurentlb) [inline build/print.go]: remove `else`

File: build/print.go [hunk 2]
```diff
@@ -631,60 +631,81 @@ const (
...
 			}
 			p.expr(x, precLow)
 		}
-
-	default:
+		// Single-element tuple must end with comma, to mark it as a tuple.
+		if len(*list) == 1 && mode == modeTuple {
+			p.printf(",")
+		}
+	} else {
```
Comment (laurentlb) [inline build/print.go]: add `return`, remove `else`

File: build/print.go (hunk 1 above)
Comment (vladmos) [reply build/print.go]: Done.

File: build/print.go (hunk 2 above)
Comment (vladmos) [reply build/print.go]: Can't return here, a closing bracket should be printed in the end of the function.

Comment (pmbethe09) [reply build/print.go]: you could return if you duplicate the code

p.depth--
if mode != modeSeq {
	p.printf("%s", brack[1:])
}

and you could de-dup by adding just after the p.depth++ above
defer func() {
  p.depth--
  if mode != modeSeq {
	p.printf("%s", brack[1:])
  }
}()

Comment (vladmos) [reply build/print.go]: Done.

**Pull request merged**